src/java/apiview-java-processor/src/main/java/com/azure/tools/apiview/processor/Main.java

```diff
@@ -167,23 +169,53 @@ private static void processJavaSourcesJar(File inputFile, APIListing apiListing)
         }
         System.out.println("  Using '" + apiListing.getLanguageVariant() + "' for the language variant");
 
-        final Analyser analyser = new JavaASTAnalyser(inputFile, apiListing);
-
-        // Read all files within the jar file so that we can create a list of files to analyse
-        final List<Path> allFiles = new ArrayList<>();
-        try (FileSystem fs = FileSystems.newFileSystem(inputFile.toPath(), Main.class.getClassLoader())) {
-            fs.getRootDirectories().forEach(root -> {
-                try (Stream<Path> paths = Files.walk(root)) {
-                    paths.forEach(allFiles::add);
-                } catch (IOException e) {
-                    e.printStackTrace();
-                    System.exit(-1);
-                }
-            });
 
-            // Do the analysis while the filesystem is still represented in memory
-            analyser.analyse(allFiles);
+        String unzippedTemp = "temp/" + inputFile.getName();
+        deleteDirectory(new File(unzippedTemp));
+
+        ZipFile zipFile = new ZipFile(inputFile);
+        zipFile.extractAll(unzippedTemp);
+        File unzippedFile = new File(unzippedTemp);
+
+        if (KotlinASTAnalyser.hasPublicApiInKotlin(unzippedFile.getAbsolutePath())) {
+            apiListing.setLanguage("Kotlin");
+            final KotlinASTAnalyser kotlinAnalyser = new KotlinASTAnalyser(apiListing);
+            kotlinAnalyser.analyse(unzippedFile.getAbsolutePath());
```

Comment:
@JonathanGiles underlying Dokka API that is used to parse Kotlin AST only takes java.io.File and that needs to be a directory. I did not find a way to virtualize that.

Reply:
I reckon you could do more or less what I have done, but rather than pass in the files individually, pass in the root directory. I would really like to see you try that, as it pains me (and I'm sure you) to see the files be unzipped in all cases!

Reply:
I do pass a root directory here `unzippedFile.getAbsolutePath()`. I do agree that it is far from ideal to unzip to the file system (not virtual one), but eventually I pass this root directory path to the Dokka, that only takes `java.io.File` which does not support virtual file system.